CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+
+<!-- omit in toc -->
+#### Before Submitting an Enhancement
+
+- Make sure that you are using the latest version.
+- Read the [documentation]() carefully and find out if the functionality is already covered, maybe by an individual configuration.
+- Perform a [search](https://github.com/semiotic-ai/autoagora-agents/issues) to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider writing an add-on/plugin library.
+
+<!-- omit in toc -->
+#### How Do I Submit a Good Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://github.com/semiotic-ai/autoagora-agents/issues).
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Provide a **step-by-step description of the suggested enhancement** in as many details as possible.
+- **Describe the current behavior** and **explain which behavior you expected to see instead** and why. At this point you can also tell which alternatives do not work for you.
+- You may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux. <!-- this should only be included if the project has a GUI -->
+- **Explain why this enhancement would be useful** to most AutoAgora agents users. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+<!-- You might want to create an issue template for enhancement suggestions that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->
+
+### Improving The Documentation
+<!-- TODO
+Updating, improving and correcting the documentation
+
+-->
+TODO
+
+## Styleguides
+
+### Code style
+
+Use [Black](https://github.com/psf/black) and [Isort](https://github.com/PyCQA/isort). Before committing code, run:
+
+```sh
+poetry run black
+poetry run isort
+```
+
+### Static type hints
+
+Use static type hints for all function inputs and outputs, as per [PEP 484](https://peps.python.org/pep-0484/).
+Check static type consistency using [Pyright](https://github.com/microsoft/pyright):
+
+```sh
+poetry run pyright
+```
+
+### Testing
+
+Use [Pytest](https://pytest.org). It is strongly encouraged to write tests for all new code. New tests for existing code are equally welcome.
```

Comment:
Do you want any level of coverage? E.g., don't submit a PR if your PR drops coverage below 90% or something? I have mixed opinions about this sort of thing as chasing coverage leads to bad testing in my experience, but without a number, I've equally found that most people won't bother to write *any* tests.

Reply:
I still have to add coverage measurement though

Reply:
Would this be a later PR then? I can mark as resolved if so

Reply:
I guess I can get it in before merging this PR.

Reply:
Fair enough. Good to not do this.